hack/sync.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source ./hack/common.sh
+
+# Please note that the validation webhooks are not currently deployed so take care when using a custom SSP CR
+export KUBEVIRT_SSP=${KUBEVIRT_SSP:-./config/samples/ssp_v1beta1_ssp.yaml}
+
+# The ssps.ssp.kubevirt.io CRD might not be present if this is the first sync so check before attempting to delete any actual objects
+if "${KUBECTL}" get crd/ssps.ssp.kubevirt.io; then
+    "${KUBECTL}" delete --ignore-not-found=true ssps -n kubevirt --all
+fi
+# The remaining CRDs should be present already from the k8s and KubeVirt installs so just ignore object not found failures
+"${KUBECTL}" delete --ignore-not-found=true deployment/ssp-operator -n kubevirt
+"${KUBECTL}" delete --ignore-not-found=true virtualmachineclusterinstancetype --all
+"${KUBECTL}" delete --ignore-not-found=true virtualmachineclusterpreference --all
+
+nodes=()
+for i in $(seq 1 "${KUBEVIRT_NUM_NODES}"); do
+    nodes+=("node$(printf "%02d" "${i}")")
+done
+
+IMG_REPOSITORY=${KUBEVIRTCI_REGISTERY}/kubevirt/ssp-operator make container-build container-push
+
+for node in ${nodes[@]}; do
+    "${KUBESSH}" "${node}" "echo registry:5000/kubevirt/ssp-operator | xargs \-\-max-args=1 sudo docker pull"
+done
+
+# TODO - clean this up, move manifest generation out of the actual tree etc.
+cd config/manager && ${KUSTOMIZE} edit set image controller=registry:5000/kubevirt/ssp-operator && cd -
```

Comment:
What about making a temp copy of the file that is changed and revert the change after kustomize ran? This helps to keep the tree clean. Maybe the existing `deploy` target in the Makefile could also be enhanced and reused?

Like here: https://github.com/0xFelix/kubevirt-addon-manager/blob/948875a264e0d75085e98f59ff2ad1c866e70603/Makefile#L55

Reply:
Yeah I might address both in a follow up if it's okay.